### PR TITLE
support transaction,  writebatch, query

### PR DESCRIPTION
**Pull request by ywmei-brt1**
support Transaction, WriteBatch, and Query

### CODE REVIEW

File: firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/IntegrationTestUtil.kt [hunk 1]
```diff
@@ -131,3 +149,10 @@ inline fun <reified T : Exception> testAssertThrows(
     val exception: T = Assert.assertThrows(T::class.java) { runnable() }
     return Truth.assertThat(exception)
 }
+
+/**
+ * makes the class can be converted to Gson JsonStr with empty constructor.
+ */
+abstract class ToGsonStringAble {
+    operator fun invoke(): String = Gson().toJson(this)
```
Comment (ywmei-brt1) [inline firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/IntegrationTestUtil.kt:157]: Can I use this to help me assert two classes' instances can be converted to the same json string?

File: firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/BatchOperationIntegrationTests.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        }
+
+    @Test
+    fun ktx_query_serialization_is_equivalent_to_java() {
+        // Test for QueryDocumentSnapshot
+        val query: Query = cities.whereEqualTo("state", "CA")
+        waitFor(query.get()).map {
+            val javaCity = it.withoutCustomMappers { toObject<JavaCity>() } as JavaCity
+            val ktxCity = it.toObject<KtxCity>()
+            assertThat(Gson().toJson(ktxCity)).isEqualTo(Gson().toJson(javaCity))
```
Comment (rlazo) [inline firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/BatchOperationIntegrationTests.kt:69]: Why do you need gson to make the comparison?

**Review by rlazo — APPROVED** [review]

**Pull request merged**